include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -449,6 +528,177 @@ namespace picongpu
+                            spanBuffer[index] = static_cast<float_X>(
+                                theTransRad[index_direction * transitionRadiation::frequencies::nOmega + k]);
+                        }
+                    }
+
+                    // Omega axis
+                    ::openPMD::Extent extentOmega
+                        = {static_cast<unsigned long int>(transitionRadiation::frequencies::nOmega), 1, 1};
+                    ::openPMD::Offset offsetOmega = {0, 0, 0};
+                    ::openPMD::Datatype datatypeOmega = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset datasetOmega{datatypeOmega, extentOmega};
+
+                    auto meshOmega = iteration.meshes["detector omega"];
+
+                    meshOmega.setAxisLabels(std::vector<std::string>{"omega", "", ""});
+                    meshOmega.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    meshOmega.setGridUnitSI(1);
+                    meshOmega.setGridSpacing(std::vector<double>{1, 1, 1});
+                    meshOmega.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+
+                    meshOmega.setUnitDimension(
+                        std::map<::openPMD::UnitDimension, double>{{::openPMD::UnitDimension::T, -1.0}});
+
+                    auto omega = meshOmega[::openPMD::RecordComponent::SCALAR];
+                    omega.resetDataset(datasetOmega);
+
+                    omega.setUnitSI(1.0);
+
+                    auto spanOmega = omega.storeChunk<float_X>(offsetOmega, extentOmega);
+                    auto spanBufferOmega = spanOmega.currentBuffer();
+
+                    for(unsigned int i = 0; i < transitionRadiation::frequencies::nOmega; ++i)
+                    {
+                        spanBufferOmega[i] = static_cast<float_X>(freqFkt(i));
+                    }
+
+                    // Phi axis
+                    ::openPMD::Extent extentPhi
+                        = {1, static_cast<unsigned long int>(transitionRadiation::parameters::nPhi), 1};
+                    ::openPMD::Offset offsetPhi = {0, 0, 0};
+                    ::openPMD::Datatype datatypePhi = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset datasetPhi{datatypePhi, extentPhi};
+
+                    auto meshPhi = iteration.meshes["detector phi"];
+
+                    meshPhi.setAxisLabels(std::vector<std::string>{"", "phi", ""});
+                    meshPhi.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    meshPhi.setGridUnitSI(1);
+                    meshPhi.setGridSpacing(std::vector<double>{1, 1, 1});
+                    meshPhi.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+
+                    auto phi = meshPhi[::openPMD::RecordComponent::SCALAR];
+                    phi.resetDataset(datasetPhi);
+
+                    phi.setUnitSI(1.0);
```

Comment:
is `phi.setUnitDimension(...)` be default all zeros?

Reply:
I thought so, but I've clarified it now in the code :)